### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by DanielSchiavini**
### What I did
Fixed the tests. Removed error swallowing.

### How I did it
By restoring the AccountDB class when resetting the env.

### How to verify it
Unit tests should be green.

### Description for the changelog
n/a

### Cute Animal Picture
image

### CODE REVIEW

File: boa/environment.py [hunk 1]
```diff
@@ -474,6 +473,9 @@ def fork_rpc(self, rpc=None, reset_traces=True, block_identifier="safe", **kwarg
     def _fork_mode(self):
         return self.vm.__class__._state_class.account_db_class == AccountDBFork
 
+    def _set_account_db_class(self, account_db_class: type):
```
Comment (DanielSchiavini) [inline boa/environment.py:476]: just moved it close to `_fork_mode`

File: boa/environment.py [hunk 2]
```diff
@@ -399,14 +399,13 @@ def set_random_seed(self, seed=None):
...
-    def _set_account_db_class(self, account_db_class: type):
-        self.vm.__class__._state_class.account_db_class = account_db_class
-
     def _init_vm(self, reset_traces=True):
         self.vm = self.chain.get_vm()
-
         self.vm.patch = VMPatcher(self.vm)
 
+        # revert any previous AccountDBFork patching
+        self._set_account_db_class(AccountDB)
```
Comment (DanielSchiavini) [inline boa/environment.py:407]: this change got lost and broke anvil, so the error did not need to be swallowed

File: Makefile [hunk 3]
```diff
@@ -3,9 +3,7 @@
 all: lint build
 
 lint:
-	black -C -t py310 boa/ tests/
-	isort boa/ tests/
-	flake8 boa/ tests/
+	pre-commit run --all-files
```
[the remark this answers is not in the document]
Comment (DanielSchiavini) [reply Makefile:6]: We can copy the one from Vyper possibly?
.github/workflows/.pre-commit-config.yaml

[the remark this answers is not in the document]
Comment (charles-cooper) [reply Makefile:6]: i think isort is just configured differently in the precommit file? see https://github.com/vyperlang/titanoboa/blob/89540c26a4a94616e45afb320829aa10a9269771/.pre-commit-config.yaml#L26

[the remark this answers is not in the document]
Comment (DanielSchiavini) [reply Makefile:6]: Yeah there used to be a mismatch between isort and black, I fixed it [here](https://github.com/vyperlang/titanoboa/commit/a2f50a887059e58aeb9c64659dabc7823bb0cae5)

[the remark this answers is not in the document]
Comment (charles-cooper) [reply Makefile:6]: ok, i guess we are getting more multi-line import statements!

[the remark this answers is not in the document]
Comment (DanielSchiavini) [reply Makefile:6]: more lines of code === more work done right? :rofl:

**Pull request merged**